alchemy/src/cloudflare/r2-rest-state-store.ts

```diff
@@ -70,6 +67,45 @@ export class R2RestStateStore implements StateStore {
     // Create Cloudflare API client with automatic account discovery
     this.api = await createCloudflareApi(this.options);
 
+    // Check if the bucket exists
+    const response = await withExponentialBackoff(
+      async () => {
+        const response = await this.api.get(
+          `/accounts/${this.api.accountId}/r2/buckets/${this.bucketName}`
+        );
+
+        if (!response.ok) {
+          await handleApiError(response, "get", "bucket", this.bucketName);
+        }
+
+        return response;
+      },
+      isRetryableError,
+      5,
+      1000
+    );
+
+    // Create the bucket if it doesn't exist
+    if (response.status === 404) {
+      await withExponentialBackoff(
+        async () => {
+          const response = await this.api.post(
+            `/accounts/${this.api.accountId}/r2/buckets`,
+            { name: this.bucketName }
+          );
```

Comment:
`createBucket` should already exist in `bucket.ts`

https://github.com/sam-goodwin/alchemy/blob/main/alchemy/src/cloudflare/bucket.ts

Reply:
Oh! Good. Didn't know if I should re-use yet. 💪

Reply:
Resolved via 4fd9cae597ecd40a617cd4bc03f84eddef5d67f0